### DISCUSSION TITLE
Welcome to JFreeSVG Discussions!

### DISCUSSION THREAD

**Discussion by jfree** [jfree/jfreesvg#40] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mickfrancis** [comment]
I would like to buy a licence for JFreeSVG, but Shopify tells me it's not possible when I go to checkout. Is this licence no longer available?

**Reply by jfree** [reply]
Hi Mick, I'm in the process of moving things away from Shopify and will instead use the Github sponsor program to fund my open source work.  Take a look on my profile page, there are sponsorship tiers where one of the rewards is a runtime license for JFreeSVG for companies that can't work with the open source license (GPLv3).  Let me know if you have any questions about it.
Best regards,
Dave

**Comment by mickfrancis** [comment]
Hi Dave,

Thanks for the reply. Probably being dumb, but where is your profile page?

Thanks,

Mick Francis
Development Manager

direct:      +44 (0) 1344 350427
mobile:   +44 7981 027106
e-mail:    ***@***.*** ***@***.***>
web:         www.lightfootsolutions.com<http://www.lightfootsolutions.com/>

Lightfoot Solutions and signalsfromnoise®(sfn) – Smarter Insights, Connected Thinking

[lightfoot_logo_blue]
Lightfoot Solutions UK Limited, Richmond House, Oldbury, Bracknell, Berkshire, RG12 8TQ.
Tel: +44 (0) 1344 350400. Fax:+44 (0) 1344 457550. Registered Office No. 06988557

Important Notice
This e-mail is confidential, may be legally privileged, and is for the intended recipient only. Access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited and may be a criminal offence. Please delete or destroy all copies immediately if obtained in error and e-mail confirmation to the sender. Thank you.

**Reply by jfree** [reply]
You should see a sponsor button here: https://github.com/jfree

**Comment by mickfrancis** [comment]
Great – thanks, Dave. Now all I need to do is break the news to Finance that the cost will be $100/month rather than £149 one-off 😊 Do you do a discount for annual sponsorship at all? I would certainly rather use JFreeSVG than Batik, which is slow and big.

Kind regards,

Mick Francis
Development Manager

direct:      +44 (0) 1344 350427
mobile:   +44 7981 027106
e-mail:    ***@***.*** ***@***.***>
web:         www.lightfootsolutions.com<http://www.lightfootsolutions.com/>

Lightfoot Solutions and signalsfromnoise®(sfn) – Smarter Insights, Connected Thinking

[lightfoot_logo_blue]
Lightfoot Solutions UK Limited, Richmond House, Oldbury, Bracknell, Berkshire, RG12 8TQ.
Tel: +44 (0) 1344 350400. Fax:+44 (0) 1344 457550. Registered Office No. 06988557

Important Notice
This e-mail is confidential, may be legally privileged, and is for the intended recipient only. Access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited and may be a criminal offence. Please delete or destroy all copies immediately if obtained in error and e-mail confirmation to the sender. Thank you.

**Reply by jfree** [reply]
The sponsorship page defaults to the recurring tiers, but there are one-off options also, so take a look in there.

**Comment by mickfrancis** [comment]
Hi Dave,

Superb – thanks!

Kind regards,

Mick Francis
Development Manager

direct:      +44 (0) 1344 350427
mobile:   +44 7981 027106
e-mail:    ***@***.*** ***@***.***>
web:         www.lightfootsolutions.com<http://www.lightfootsolutions.com/>

Lightfoot Solutions and signalsfromnoise®(sfn) – Smarter Insights, Connected Thinking

[lightfoot_logo_blue]
Lightfoot Solutions UK Limited, Richmond House, Oldbury, Bracknell, Berkshire, RG12 8TQ.
Tel: +44 (0) 1344 350400. Fax:+44 (0) 1344 457550. Registered Office No. 06988557

Important Notice
This e-mail is confidential, may be legally privileged, and is for the intended recipient only. Access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited and may be a criminal offence. Please delete or destroy all copies immediately if obtained in error and e-mail confirmation to the sender. Thank you.

**Comment by mickfrancis** [comment]
Hi Dave,

Sorry to bother you, but I have one further question: What is the mechanism for granting the Runtime Licence? I will need it to be registered with the company I work for, not for me. Does the company need to have its own GitHub account?

Kind regards,

Mick Francis
Development Manager

direct:      +44 (0) 1344 350427
mobile:   +44 7981 027106
e-mail:    ***@***.*** ***@***.***>
web:         www.lightfootsolutions.com<http://www.lightfootsolutions.com/>

Lightfoot Solutions and signalsfromnoise®(sfn) – Smarter Insights, Connected Thinking

[lightfoot_logo_blue]
Lightfoot Solutions UK Limited, Richmond House, Oldbury, Bracknell, Berkshire, RG12 8TQ.
Tel: +44 (0) 1344 350400. Fax:+44 (0) 1344 457550. Registered Office No. 06988557

Important Notice
This e-mail is confidential, may be legally privileged, and is for the intended recipient only. Access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited and may be a criminal offence. Please delete or destroy all copies immediately if obtained in error and e-mail confirmation to the sender. Thank you.

**Comment by mickfrancis** [comment]
Hi Dave,

We’ve completed the one-time $100 sponsorship (user LightfootReception). I have 2 questions:


  *   How do we go about getting the Runtime Licence for JFreeSVG?
  *   How do we access the developer materials? When I go to https://www.jfree.org/sponsors/index.html I get:
    ***@***.***

Thanks for the great software – very impressive!

Kind regards,

Mick Francis
Development Manager

direct:      +44 (0) 1344 350427
mobile:   +44 7981 027106
e-mail:    ***@***.*** ***@***.***>
web:         www.lightfootsolutions.com<http://www.lightfootsolutions.com/>

Lightfoot Solutions and signalsfromnoise®(sfn) – Smarter Insights, Connected Thinking

[lightfoot_logo_blue]
Lightfoot Solutions UK Limited, Richmond House, Oldbury, Bracknell, Berkshire, RG12 8TQ.
Tel: +44 (0) 1344 350400. Fax:+44 (0) 1344 457550. Registered Office No. 06988557

Important Notice
This e-mail is confidential, may be legally privileged, and is for the intended recipient only. Access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited and may be a criminal offence. Please delete or destroy all copies immediately if obtained in error and e-mail confirmation to the sender. Thank you.



From: Mick Francis
Sent: 28 February 2022 12:37
To: jfree/jfreesvg ***@***.***>
Subject: RE: [jfree/jfreesvg] Welcome to JFreeSVG Discussions! (Discussion #40)

Hi Dave,

Sorry to bother you, but I have one further question: What is the mechanism for granting the Runtime Licence? I will need it to be registered with the company I work for, not for me. Does the company need to have its own GitHub account?

Kind regards,

Mick Francis
Development Manager

direct:      +44 (0) 1344 350427
mobile:   +44 7981 027106
e-mail:    ***@***.*** ***@***.***>
web:         www.lightfootsolutions.com<http://www.lightfootsolutions.com/>

Lightfoot Solutions and signalsfromnoise®(sfn) – Smarter Insights, Connected Thinking

[lightfoot_logo_blue]
Lightfoot Solutions UK Limited, Richmond House, Oldbury, Bracknell, Berkshire, RG12 8TQ.
Tel: +44 (0) 1344 350400. Fax:+44 (0) 1344 457550. Registered Office No. 06988557

Important Notice
This e-mail is confidential, may be legally privileged, and is for the intended recipient only. Access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited and may be a criminal offence. Please delete or destroy all copies immediately if obtained in error and e-mail confirmation to the sender. Thank you.

**Reply by jfree** [reply]
Hi Mick,

Thanks for being the first Github sponsor for JFree!  It's possible I haven't configured the email correctly, can you forward what you received to dave@jfree.org?  In the meantime I'll send you the license and instructions to access the other materials.

Best regards,
Dave

**Comment by mickfrancis** [comment]
Hi Dave,

I hope you don’t mind me contacting you. Back in Feb 2022 we became a sponsor of JFreeSVG. It’s still performing really well for us, so thanks once again!

There are 2 issues, though, that you may be interested in:


  1.  Size of SVG
We post-process the SVG to reduce the size by 20-30% (10-15% when zipped), using simple transformations such as removing trailing “.0” from decimals, replacing long ids with shorter ones, removing values if they are defaults, using shorter transform names. E.g.:

***@***.***
This takes around 100ms for a 0.5Mb SVG, which while not excessive, we would prefer not to have to do. Might you consider adding in these kind of optimisations? It would further widen the gap between JFreeSVG and Batik 😊

  2.  Slightly more pressing there is an issue we have when drawing table headers (not necessarily related to tables though):

The Java Graphics2D produces:
    ***@***.***
whereas in JFreeSVG the header titles get truncated:
    ***@***.***
The code auto-fits, so the titles should fit.

I appreciate that without a test case this would be hard to address, but just wondered if this is a known issue at all? It seems to affect strings with spaces.

Have a great Christmas,

Mick Francis
Development Manager

direct:    +44 (0) 1344 350427
mobile:  +44 7981 027106
e-mail:    ***@***.*** ***@***.***>
web:        www.lightfootsolutions.com<http://www.lightfootsolutions.com/>

Lightfoot Solutions and signalsfromnoise®(sfn) – Smarter Insights, Connected Thinking

[lightfoot_logo_blue]
Lightfoot Solutions UK Limited, Richmond House, Oldbury, Bracknell, Berkshire, RG12 8TQ.
Tel: +44 (0) 1344 350400. Fax:+44 (0) 1344 457550. Registered Office No. 06988557

Important Notice
This e-mail is confidential, may be legally privileged, and is for the intended recipient only. Access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited and may be a criminal offence. Please delete or destroy all copies immediately if obtained in error and e-mail confirmation to the sender. Thank you.



From: Mick Francis
Sent: 28 February 2022 16:33
To: 'jfree/jfreesvg' ***@***.***>
Subject: RE: [jfree/jfreesvg] Welcome to JFreeSVG Discussions! (Discussion #40)

Hi Dave,

We’ve completed the one-time $100 sponsorship (user LightfootReception). I have 2 questions:


  *   How do we go about getting the Runtime Licence for JFreeSVG?
  *   How do we access the developer materials? When I go to https://www.jfree.org/sponsors/index.html I get:
    ***@***.***

Thanks for the great software – very impressive!

Kind regards,

Mick Francis
Development Manager

direct:      +44 (0) 1344 350427
mobile:   +44 7981 027106
e-mail:    ***@***.*** ***@***.***>
web:         www.lightfootsolutions.com<http://www.lightfootsolutions.com/>

Lightfoot Solutions and signalsfromnoise®(sfn) – Smarter Insights, Connected Thinking

[lightfoot_logo_blue]
Lightfoot Solutions UK Limited, Richmond House, Oldbury, Bracknell, Berkshire, RG12 8TQ.
Tel: +44 (0) 1344 350400. Fax:+44 (0) 1344 457550. Registered Office No. 06988557

Important Notice
This e-mail is confidential, may be legally privileged, and is for the intended recipient only. Access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited and may be a criminal offence. Please delete or destroy all copies immediately if obtained in error and e-mail confirmation to the sender. Thank you.



From: Mick Francis
Sent: 28 February 2022 12:37
To: jfree/jfreesvg ***@***.******@***.***>>
Subject: RE: [jfree/jfreesvg] Welcome to JFreeSVG Discussions! (Discussion #40)

Hi Dave,

Sorry to bother you, but I have one further question: What is the mechanism for granting the Runtime Licence? I will need it to be registered with the company I work for, not for me. Does the company need to have its own GitHub account?

Kind regards,

Mick Francis
Development Manager

direct:      +44 (0) 1344 350427
mobile:   +44 7981 027106
e-mail:    ***@***.*** ***@***.***>
web:         www.lightfootsolutions.com<http://www.lightfootsolutions.com/>

Lightfoot Solutions and signalsfromnoise®(sfn) – Smarter Insights, Connected Thinking

[lightfoot_logo_blue]
Lightfoot Solutions UK Limited, Richmond House, Oldbury, Bracknell, Berkshire, RG12 8TQ.
Tel: +44 (0) 1344 350400. Fax:+44 (0) 1344 457550. Registered Office No. 06988557

Important Notice
This e-mail is confidential, may be legally privileged, and is for the intended recipient only. Access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited and may be a criminal offence. Please delete or destroy all copies immediately if obtained in error and e-mail confirmation to the sender. Thank you.

**Comment by mickfrancis** [comment]
P.S. Attached is an example of before and after ion case that helps.

Mick Francis
Development Manager

direct:    +44 (0) 1344 350427
mobile:  +44 7981 027106
e-mail:    ***@***.*** ***@***.***>
web:        www.lightfootsolutions.com<http://www.lightfootsolutions.com/>

Lightfoot Solutions and signalsfromnoise®(sfn) – Smarter Insights, Connected Thinking

[lightfoot_logo_blue]
Lightfoot Solutions UK Limited, Richmond House, Oldbury, Bracknell, Berkshire, RG12 8TQ.
Tel: +44 (0) 1344 350400. Fax:+44 (0) 1344 457550. Registered Office No. 06988557

Important Notice
This e-mail is confidential, may be legally privileged, and is for the intended recipient only. Access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited and may be a criminal offence. Please delete or destroy all copies immediately if obtained in error and e-mail confirmation to the sender. Thank you.



From: Mick Francis
Sent: 20 December 2024 13:51
To: jfree/jfreesvg ***@***.***>
Subject: RE: [jfree/jfreesvg] Welcome to JFreeSVG Discussions! (Discussion #40)

Hi Dave,

I hope you don’t mind me contacting you. Back in Feb 2022 we became a sponsor of JFreeSVG. It’s still performing really well for us, so thanks once again!

There are 2 issues, though, that you may be interested in:


  1.  Size of SVG
We post-process the SVG to reduce the size by 20-30% (10-15% when zipped), using simple transformations such as removing trailing “.0” from decimals, replacing long ids with shorter ones, removing values if they are defaults, using shorter transform names. E.g.:

***@***.***
This takes around 100ms for a 0.5Mb SVG, which while not excessive, we would prefer not to have to do. Might you consider adding in these kind of optimisations? It would further widen the gap between JFreeSVG and Batik 😊
  2.  Slightly more pressing there is an issue we have when drawing table headers (not necessarily related to tables though):

The Java Graphics2D produces:
    ***@***.***
whereas in JFreeSVG the header titles get truncated:
    ***@***.***
The code auto-fits, so the titles should fit.

I appreciate that without a test case this would be hard to address, but just wondered if this is a known issue at all? It seems to affect strings with spaces.

Have a great Christmas,

Mick Francis
Development Manager

direct:    +44 (0) 1344 350427
mobile:  +44 7981 027106
e-mail:    ***@***.*** ***@***.***>
web:        www.lightfootsolutions.com<http://www.lightfootsolutions.com/>

Lightfoot Solutions and signalsfromnoise®(sfn) – Smarter Insights, Connected Thinking

[lightfoot_logo_blue]
Lightfoot Solutions UK Limited, Richmond House, Oldbury, Bracknell, Berkshire, RG12 8TQ.
Tel: +44 (0) 1344 350400. Fax:+44 (0) 1344 457550. Registered Office No. 06988557

Important Notice
This e-mail is confidential, may be legally privileged, and is for the intended recipient only. Access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited and may be a criminal offence. Please delete or destroy all copies immediately if obtained in error and e-mail confirmation to the sender. Thank you.



From: Mick Francis
Sent: 28 February 2022 16:33
To: 'jfree/jfreesvg' ***@***.******@***.***>>
Subject: RE: [jfree/jfreesvg] Welcome to JFreeSVG Discussions! (Discussion #40)

Hi Dave,

We’ve completed the one-time $100 sponsorship (user LightfootReception). I have 2 questions:


  *   How do we go about getting the Runtime Licence for JFreeSVG?
  *   How do we access the developer materials? When I go to https://www.jfree.org/sponsors/index.html I get:
    ***@***.***

Thanks for the great software – very impressive!

Kind regards,

Mick Francis
Development Manager

direct:      +44 (0) 1344 350427
mobile:   +44 7981 027106
e-mail:    ***@***.*** ***@***.***>
web:         www.lightfootsolutions.com<http://www.lightfootsolutions.com/>

Lightfoot Solutions and signalsfromnoise®(sfn) – Smarter Insights, Connected Thinking

[lightfoot_logo_blue]
Lightfoot Solutions UK Limited, Richmond House, Oldbury, Bracknell, Berkshire, RG12 8TQ.
Tel: +44 (0) 1344 350400. Fax:+44 (0) 1344 457550. Registered Office No. 06988557

Important Notice
This e-mail is confidential, may be legally privileged, and is for the intended recipient only. Access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited and may be a criminal offence. Please delete or destroy all copies immediately if obtained in error and e-mail confirmation to the sender. Thank you.



From: Mick Francis
Sent: 28 February 2022 12:37
To: jfree/jfreesvg ***@***.******@***.***>>
Subject: RE: [jfree/jfreesvg] Welcome to JFreeSVG Discussions! (Discussion #40)

Hi Dave,

Sorry to bother you, but I have one further question: What is the mechanism for granting the Runtime Licence? I will need it to be registered with the company I work for, not for me. Does the company need to have its own GitHub account?

Kind regards,

Mick Francis
Development Manager

direct:      +44 (0) 1344 350427
mobile:   +44 7981 027106
e-mail:    ***@***.*** ***@***.***>
web:         www.lightfootsolutions.com<http://www.lightfootsolutions.com/>

Lightfoot Solutions and signalsfromnoise®(sfn) – Smarter Insights, Connected Thinking

[lightfoot_logo_blue]
Lightfoot Solutions UK Limited, Richmond House, Oldbury, Bracknell, Berkshire, RG12 8TQ.
Tel: +44 (0) 1344 350400. Fax:+44 (0) 1344 457550. Registered Office No. 06988557

Important Notice
This e-mail is confidential, may be legally privileged, and is for the intended recipient only. Access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited and may be a criminal offence. Please delete or destroy all copies immediately if obtained in error and e-mail confirmation to the sender. Thank you.
